plugins/modules/podman_prune.py

```diff
@@ -0,0 +1,247 @@
+          until: 24h
+
+- name: Prune everything
+  containers.podman.podman_prune:
+      system: true
+
+- name: Prune everything (including non-dangling images)
+  containers.podman.podman_prune:
+      system: true
+      system_all: true
+      system_volumes: true
+'''
+
+RETURN = r'''
+# containers
+containers:
+    description:
+      - List of IDs of deleted containers.
+    returned: I(containers) is C(true)
+    type: list
+    elements: str
+    sample: []
+
+# images
+images:
+    description:
+      - List of IDs of deleted images.
+    returned: I(images) is C(true)
+    type: list
+    elements: str
+    sample: []
+
+# networks
+networks:
+    description:
+      - List of IDs of deleted networks.
+    returned: I(networks) is C(true)
+    type: list
+    elements: str
+    sample: []
+
+# volumes
+volumes:
+    description:
+      - List of IDs of deleted volumes.
+    returned: I(volumes) is C(true)
+    type: list
+    elements: str
+    sample: []
+
+# system
+system:
+  description:
+    - List of ID of deleted containers, volumes, images, network and total reclaimed space
+  returned: I(system) is C(true)
+  type: list
+  elements: str
+  sample: []
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def podmanExec(module, target, filters, executable):
+    command = [executable, target, 'prune', '--force']
+    if filters:
+        if target != 'system':
+            for common_filter in filters:
+                if isinstance(filters[common_filter], dict):
+                    dict_filters = filters[common_filter]
+                    for single_filter in dict_filters:
+                        command.append('--filter={label}={key}={value}'.format(label=common_filter, key=single_filter,
+                                                                               value=dict_filters[single_filter]))
+                else:
+                    if target == 'image' and (common_filter in ('dangling_only', 'external')):
+                        if common_filter == 'dangling_only' and not filters['dangling_only']:
+                            command.append('-a')
+                        if common_filter == 'external' and filters['external']:
+                            command.append('--external')
+                    else:
+                        command.append('--filter={label}={value}'.format(label=common_filter,
+                                                                         value=filters[common_filter]))
+        else:
+            for system_filter in filters:
+                command.append(filters[system_filter])
+
+    rc, out, err = module.run_command(command)
+    if out:
+        changed = True
+    else:
+        changed = False
+    if rc != 0:
+        module.fail_json(
+            msg='Error executing prune on {target}: {err}'.format(target=target, err=err))
+
+    return {
+        "changed": changed,
+        "stdout": out,
+        "stderr": err
+    }
```

Comment:
Actually this part should be in podman_exec function, filter prepare in a separate.